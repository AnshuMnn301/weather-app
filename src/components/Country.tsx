import React from 'react';
import {View, Image, Text, Button} from 'react-native';
interface WeatherType {
  weather_icons: Array<string>;
  temperature: string;
  wind_speed: string;
  precip: string;
}
export default function Country(props: any) {
  const [weatherData, setWatherData] = React.useState<WeatherType | null>(null);
  const handleGetWeather = async () => {
    const resp = await fetch(
      `http://api.weatherstack.com/current?access_key=8c483281e15083c4d1a2bb16bccf1736&query=${props.capital}`,
    );
    const data = await resp.json();
    setWatherData(data?.current);
  };
  return (
    <View
      style={{
        backgroundColor: 'silver',
        paddingTop: 10,
        alignItems: 'center',
        borderTopEndRadius: 10,
        borderTopStartRadius: 10,
        marginVertical: 5,
        width: '95%',
        marginLeft: 10,
      }}>
      <Image
        source={{uri: props.imageSource}}
        style={{width: '100%', height: 100, resizeMode: 'contain'}}
      />
      <Text style={{marginVertical: 5, fontWeight: 'bold'}}>{props.title}</Text>
      <Button title="Get Weather Forecast" onPress={handleGetWeather} />
      <View style={{paddingBottom: 10}} />
      {weatherData !== null && (
        <View
          style={{
            paddingVertical: 10,
            flexDirection: 'row',
            alignItems: 'center',
            justifyContent: 'center',
            backgroundColor: 'lightgray',
            width: '100%',
          }}>
          <Image
            source={{uri: weatherData?.weather_icons[0]}}
            style={{width: 50, height: 50}}
          />
          <View style={{marginLeft: 15}}>
            <Text>
              Temprature: {weatherData.temperature} {'\n'}Precipitation:
              {weatherData.precip}
              {'\n'}Wind speed: {weatherData.wind_speed}
            </Text>
          </View>
        </View>
      )}
    </View>
  );
}
