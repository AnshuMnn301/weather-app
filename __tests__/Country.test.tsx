/**
 * @format
 */
import {Button, Text} from 'react-native'
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer, { act } from 'react-test-renderer';
import Country from '../src/components/Country';
import {shallow } from 'enzyme';
import fetchMock from 'fetch-mock-jest'

test('render Country Component corrcetly', () => {
    const tree = renderer.create(<Country/>).toJSON()
    expect(tree).toMatchSnapshot()
})

describe('Test Country component', () => {
    let wrapper
    afterEach(() => {
        fetchMock.mockReset()
    })
    beforeEach(() => {
        const props =  {"imageSource": "https://flagcdn.com/w320/us.png", "latlng": [38.89, -77.05], "title": "United States", "capital": "Washington, D.C."}
        fetchMock.get(`http://api.weatherstack.com/current?access_key=8c483281e15083c4d1a2bb16bccf1736&query=${props.capital}`,{"cloudcover": 75, "feelslike": 39, "humidity": 67, "is_day": "yes", "observation_time": "04:36 PM", "precip": 0, "pressure": 1013, "temperature": 32, "uv_index": 8, "visibility": 10, "weather_code": 116, "weather_descriptions": ["Partly cloudy"], "weather_icons": ["https://cdn.worldweatheronline.com/images/wsymbols01_png_64/wsymbol_0002_sunny_intervals.png"], "wind_degree": 100, "wind_dir": "E", "wind_speed": 28})
        wrapper = shallow(<Country {...props}/>)
    })
    it('Test getting weather data', async() => {
       
        const getWeatherButton = wrapper.find(Button)
        getWeatherButton.props().onPress();
        await act(() => new Promise(setImmediate));
        wrapper.update()
    })
})