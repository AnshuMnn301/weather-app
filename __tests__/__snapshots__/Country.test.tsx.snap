// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Country Component corrcetly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "silver",
      "borderTopEndRadius": 10,
      "borderTopStartRadius": 10,
      "marginLeft": 10,
      "marginVertical": 5,
      "paddingTop": 10,
      "width": "95%",
    }
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "height": 100,
        "resizeMode": "contain",
        "width": "100%",
      }
    }
  />
  <Text
    style={
      Object {
        "fontWeight": "bold",
        "marginVertical": 5,
      }
    }
  />
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Get Weather Forecast
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "paddingBottom": 10,
      }
    }
  />
</View>
`;
